src/index.js

```diff
@@ -2,10 +2,13 @@ import isTextLike from './is-text-like';
 import { isHtmlVoidTag, isSvgVoidTag } from './void-tags';
 import parseQuery from './parse-query';
 
+const queryCache = {};
+
 export default function tag (query) {
+  query = (queryCache[query] ? queryCache[query] : (query = queryCache[query] = parseQuery(query)));
```

Comment:
Should argument reassigning be avoided though?
https://spin.atomicobject.com/2011/04/10/javascript-don-t-reassign-your-function-arguments/

Reply:
> The point of this post is to raise awareness that reassigning the value of an argument variable mutates the `arguments` object. 

We aren't using `arguments` here. _Mutating_ the internals of `query` would be a no-no.

Other than that, it's just a stylistic/linty thing, same as using `const` when possible (since arguments are in this case practically `var`-bindings).

Reply:
Yeah, I'd just prefer:

```js
function (_query) {
  const query = _query ...
```